### PR TITLE
LIVY-326. Change Hadoop dependencies to Apache Hadoop

### PR DESCRIPTION
**Pull request by jerryshao**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -42,7 +42,7 @@
   </organization>
 
   <properties>
-    <hadoop.version>2.6.0-cdh5.5.0</hadoop.version>
+    <hadoop.version>2.6.5</hadoop.version>
```
Comment (zjffdu) [inline pom.xml]: How about change it to 2.7.x which has been released for a while

Comment (jerryshao) [inline pom.xml]: Yes, personally I don't have a strong preference on which version of Hadoop to use, I just keep align with Spark's minimum supported version of Hadoop.

**Comment by jerryshao** [comment]
@alex-the-man do you have any suggestion on which version of Hadoop should be used?

**Comment by jerryshao** [comment]
Looks there's a unrelated flaky test.

**Comment by alex-the-man** [comment]
Since we don't have multiple profiles like Spark has in their pom.xml, I think it's better to use 2.7 so we are running our integration test with the most popular YARN version.

**Comment by zjffdu** [comment]
LGTM, wait for CI pass

**Pull request merged**